src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -256,23 +256,29 @@ private static void ThrowServerException(HttpResponse response, RequestContext r
                 exceptionToThrow = MsalServiceExceptionFactory.FromHttpResponse(
                     MsalError.NonParsableOAuthError,
                     MsalErrorMessage.NonParsableOAuthError,
-                    response);
+                    response,
+                    null,
+                    requestContext);
             }
             catch (Exception ex)
             {
-
                 exceptionToThrow = MsalServiceExceptionFactory.FromHttpResponse(
                     MsalError.UnknownError,
                     response.Body,
                     response,
-                    ex);
+                    ex,
+                    requestContext);
             }
 
+            bool isCodeNotFound = response.StatusCode == HttpStatusCode.NotFound;
+
             exceptionToThrow ??= MsalServiceExceptionFactory.FromHttpResponse(
```

Comment:
THis is hard to read. Why don't you just do a `if (response.StatusCode == HttpStatusCode.NotFound) { throw e1 } else { throw e2}